web/controllers/api/key_controller.ex

```diff
@@ -23,7 +23,18 @@ defmodule HexWeb.API.KeyController do
   end
 
   def create(conn, params) do
-    case Key.create(conn.assigns.user, conn.params) |> HexWeb.Repo.insert do
+    result =
+      HexWeb.Repo.transaction(fn ->
+        case Key.create(conn.assigns.user, conn.params) |> HexWeb.Repo.insert do
+          {:ok, key} ->
+            HexWeb.AuditLog.create(conn.assigns.user, "key.generate", key) |> HexWeb.Repo.insert!
```

Comment:
@ericmj I pushed the changes with rollback as we discussed. Do you think it's OK to do Repo.insert! for audit log, or it would be better to handle possible error here as well? I don't really expect this to ever fail other than when the DB is offline or full.

cc @jcspencer